src/core/level.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * This module updates the title and shortName
+ * of a spec when there is a valid level configuration.
+ *
+ * Levels should be integers >= 0.
+ */
+
+import { getIntlData, showInlineWarning } from "./utils.js";
+export const name = "core/level";
+
+const localizationStrings = {
+  en: {
+    level: "Level",
+  },
+};
+
+const l10n = getIntlData(localizationStrings);
+
+export function run(conf) {
+  const h1Elem = document.querySelector("h1#title");
+
+  if (conf.hasOwnProperty("level")) {
+    if (Number.isInteger(conf.level) && conf.level >= 0) {
+      h1Elem.append(` ${l10n.level} ${conf.level}`);
+      document.title = `${document.title} ${l10n.level} ${conf.level}`;
+      conf.shortName = `${conf.shortName}-${conf.level}`;
```

Comment:
I'm a bit concerned that this might break `data-cite` at:
https://github.com/w3c/respec/blob/63fd45b24125edd7795375809800b89d9001d36b/src/core/data-cite.js#L33-L39

Can't find any existing test or use-case for this though

Reply:
hm. let's take [navigation timing level 2](https://w3c.github.io/navigation-timing/) as an example.

their shortName is `navigation-timing-2`.

like most specs, that spec relies on other specs and cites them by their shortNames. eg:
```
      xref: {
        url: "https://respec.org/xref",
        specs: [ "service-workers", "hr-time-2", "performance-timeline-2", "resource-timing-2"],
        profile: "web-platform",
      },
```

it seems that if we rely on respec to append the level number to the shortName, we'd remove the `-2` from those specs. respec would then be confused when there's `spec-1` and `spec-2`

Reply:
The xref part is fine here, and it should use numbered shortnames. I think there might be a problem with "self citing". Like, we can't do `[[navigation-timing]]`. Not sure if it's used though.
Let me find some examples. Otherwise, we can remove that part from data-cite.js itself

Reply:
Apparently, we won't be able to link stuff like:
``` html
    <section id="test">
      <h2>test</h2>
      <a data-cite="shortname-api#test">link to test</a>
    </section>
``` 
which is good as we want to discourage such usage!

Reply:
cc: @marcoscaceres who might know why we have such condition in data-cite.js

Reply:
> The xref part is fine here, and it should use numbered shortnames

hm. my concern is that it using the level config would mean non-numbered shortnames.  as you can see from the [requirements](https://github.com/w3c/respec/issues/2406) in the original issue, 

> combines the level with the shortName (so... page-visibility becomes "page-visibility-2")

Thus, a spec using the levels feature would have `config = { shortName: `navigation-timing`, level: 2` because respec would automatically append `-2` to `navigation-timing`. my understanding of the code that you linked is that the lack of a numbered shortName might break things.

@marcoscaceres could you help us clarify? thanks ^^

Reply:
Specs that self citing itself is "nonsensical": it doesn't make any sense to ever do that, specially in light of xref and the HTML's self linking capability. So I don't think we should worry about it.

Reply:
Just for reference, the un-numbered short name always points to the highest level. For example:
https://www.w3.org/TR/css-color/  <- is, today, an alias for "css-color-4". 

Similarly, these all exist:
ttps://www.w3.org/TR/css-color-4/
https://www.w3.org/TR/css-color-3/
https://www.w3.org/TR/css-color-2/

So, by W3C policy, we should be fine*.

<small>*Terms and conditions apply. May cause irreversible damage to the liver and kidneys. Please consult your doctor before using levels.</small>.

Reply:
Thanks for the explanation and background info both here and in slack @marcoscaceres! It's a lot clearer now.